pkg/controller/appsync/appsync.go

```diff
@@ -44,8 +49,98 @@ func New(config *config.AppSyncConfig, db *database.Database, h *render.Renderer
 
 // HandleSync performs the logic to sync mobile apps.
 func (c *Controller) HandleSync(ctx context.Context) http.Handler {
+	type AppSyncResult struct {
+		OK     bool    `json:"ok"`
+		Errors []error `json:"errors,omitempty"`
+	}
+
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		// TODO(whaught): implement this
-		controller.InternalError(w, r, c.h, errors.New("not implemented"))
+		logger := logging.FromContext(ctx).Named("appsync.HandleSync")
+
+		client := http.Client{Timeout: c.config.Timeout}
+		resp, err := client.Get(c.config.AppSyncURL)
+		if err != nil {
+			controller.InternalError(w, r, c.h, err)
+			return
+		}
+
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(io.LimitReader(resp.Body, c.config.FileSizeLimit))
+		if err != nil {
+			controller.InternalError(w, r, c.h, err)
+			return
+		}
+
+		var apps AppsResponse
+		if err := json.Unmarshal(body, &apps); err != nil {
+			controller.InternalError(w, r, c.h, err)
+			return
+		}
+
+		var merr *multierror.Error
+		realms := map[string]*database.Realm{}
+		appsByRealm := map[uint][]*database.MobileApp{}
+
+		for _, app := range apps.Apps {
+			realm, has := realms[app.Region]
+			if !has {
+				realm, err := c.db.FindRealmByRegion(app.Region)
+				if err != nil {
+					merr = multierror.Append(merr, fmt.Errorf("unable to lookup realm: %w", err))
+					continue
+				}
+				realms[app.Region] = realm
+			}
+
+			realmApps, has := appsByRealm[realm.ID]
+			if !has {
+				// Only Android supported for sync.
+				realmApps, err := c.db.ListActiveAppsByOS(realm.ID, database.OSTypeAndroid)
+				if err != nil {
+					merr = multierror.Append(merr, fmt.Errorf("unable to lookup realm: %w", err))
+					continue
+				}
+				appsByRealm[realm.ID] = realmApps
+			}
+
+			has = false
```

Comment:
I'm not sure I follow what this is doing?

Reply:
Sorry I'm kind of working in the open here (this is not ready). I'm trying to match the given JSON app to the existing `MobileApp` (per realm for android). At the moment I'm matching on the SHA in that list, but trying to understand what `package_name` maps to how how to set Name, URL, and AppID

Reply:
So I probably did a useless thing in having the 'target' field mirror the content that will end up at https://www.example.com/.well-known/assetlinks.json:
https://developer.android.com/training/app-links/verify-site-associations

But presumably you care more about mapping to existing config fields.

package_name == AppID
URL === https://play.google.com/store/apps/details?id=<package_name>
e.g. https://play.google.com/store/apps/details?id=gov.co.cdphe.exposurenotifications
If you'd prefer, I can add a field for the URL explicitly in the generated json?

For Name, I'm unsure, we have the app name for various locales, so I could give you the default locale, but I had assumed we didn't need to check these match, as the PHA might configure the device key once and then rename their app and not want to keep both in sync? WDYT?

Reply:
I've added in that with the a default name "[package_name] Android App", but it's not my favorite.
The more that's in the published file, the less the server needs to guess / hardcode, so that's my preference.